tests/functional/test_chain.py

```diff
@@ -219,3 +227,136 @@ def test_instance_at_when_given_name_as_contract_type(chain, contract_instance):
+    deployed_contract_1 = owner.deploy(project_with_contract.ApeContract1, required_confirmations=0)
+    deployments_mapping = chain.contracts._load_deployments_mapping()
+
+    # Act
+    my_contracts_list_0 = chain.contracts.get_deployments(project_with_contract.ApeContract0)
+    my_contracts_list_1 = chain.contracts.get_deployments(project_with_contract.ApeContract1)
+
+    # Assert
+    assert (
+        deployments_mapping["ethereum"]["rinkeby"]["ApeContract0"][-1]
+        == deployed_contract_0.address
+    )
+    assert my_contracts_list_0[-1].address == deployed_contract_0.address
+    assert (
+        deployments_mapping["ethereum"]["rinkeby"]["ApeContract1"][-1]
+        == deployed_contract_1.address
+    )
+    assert my_contracts_list_1[-1].address == deployed_contract_1.address
+
+
+def test_get_multiple_deployments_live(
+    chain, project_with_contract, owner, remove_disk_writes_deployments, dummy_live_network
+):
+    # Arrange
+    initial_deployed_contract_0 = owner.deploy(
+        project_with_contract.ApeContract0, required_confirmations=0
+    )
+    initial_deployed_contract_1 = owner.deploy(
+        project_with_contract.ApeContract1, required_confirmations=0
+    )
+    owner.deploy(project_with_contract.ApeContract0, required_confirmations=0)
+    owner.deploy(project_with_contract.ApeContract1, required_confirmations=0)
+    final_deployed_contract_0 = owner.deploy(
+        project_with_contract.ApeContract0, required_confirmations=0
+    )
+    final_deployed_contract_1 = owner.deploy(
+        project_with_contract.ApeContract1, required_confirmations=0
+    )
+    deployments_mapping = chain.contracts._load_deployments_mapping()
+
+    # Act
+    contracts_list_0 = chain.contracts.get_deployments(project_with_contract.ApeContract0)
+    contracts_list_1 = chain.contracts.get_deployments(project_with_contract.ApeContract1)
+
+    # Assert
+    assert (
+        deployments_mapping["ethereum"]["rinkeby"]["ApeContract0"][0]
+        == initial_deployed_contract_0.address
+    )
+    assert contracts_list_0[-1].address == final_deployed_contract_0.address
+    assert len(contracts_list_0) == 3
+    assert (
+        deployments_mapping["ethereum"]["rinkeby"]["ApeContract1"][0]
+        == initial_deployed_contract_1.address
+    )
+    assert contracts_list_1[-1].address == final_deployed_contract_1.address
+    assert len(contracts_list_1) == 3
+
+
+def test_contract_cache_mapping_updated_on_many_deployments(owner, project_with_contract, chain):
+    # Arrange / Act
+    initial_deployed_contract = owner.deploy(project_with_contract.ApeContract0)
+    owner.deploy(project_with_contract.ApeContract0)
+    owner.deploy(project_with_contract.ApeContract0)
+    final_deployed_contract = owner.deploy(project_with_contract.ApeContract0)
+
+    my_contracts_list = chain.contracts.get_deployments(project_with_contract.ApeContract0)
+
+    # Assert
+    assert len(my_contracts_list) == 4
```

Comment:
Either need to assert that this is 4 + more than before running the test (like grab the start count in the Arrange section)..

OR

use the fixture that clears the cache before + after.